### DISCUSSION TITLE
How to log a dynamic string

### DISCUSSION THREAD

**Discussion by diehard2** [odygrd/quill#382] [General]
Sorry if this is obvious, but I've been banging my head against a wall trying to figure it out

with fmt I can do 

```
std::string message("foo {}")
auto var = fmt::format(fmt::runtime(message), 1);
```

and it will work

if I try

LOG_DEBUG(loggerp, message, 1) or LOG_DEBUG(loggerp, fmt::runtime(message), 1)

I'm getting 

 error: use of local variable with automatic storage from containing function. 

The end goal is to log a dynamic string. Really great library btw!

**Comment by odygrd** [comment]
hey, 

dynamic format string is not supported, it has to be specified in compile time as the library is creating constexpr objects underneath. 

You can simulate it, something like this for example :

```c++
std::string message("foo {}")
auto var = fmt::format(fmt::runtime(message), 1);
LOG_DEBUG(loggerp, "{}", var)
```

You can also look at related https://github.com/odygrd/quill/issues/361

This eventually is adding formatting overhead on the hot path rather than the backend logging thread, but it is not easy to support dynamic format strings
